sounddevice.py

```diff
@@ -2262,13 +2262,67 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : str, optional
+        User error message.
+    code : int, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
+
+    """
+
+    code = None
+    """The PaErrorCode integer value or None if there is no associated error
+    code.
+
+    """
+    host_api = None
+    """The index of the host API which returned the error code or None if the
+    error is not a host API error.
+
+    """
+    hosterror_info = None
+    """Either a tuple containing the host error code and a textual description
+    of the host error or None if the error is not a host API error. The host
+    error description will be a zero length string if one is not available.
+
+    """
+
+    def __init__(self, msg=None, code=None):
+        self.code = code
+        self.hosterror_info = self.host_api = None
```

Comment:
You don't need to set them to `None` because the class variables are already there.